lib/Conversion/HandshakeToFIRRTL/HandshakeToFIRRTL.cpp

```diff
@@ -801,6 +799,103 @@ static void convertReturnOp(Operation *oldOp, FModuleOp topModuleOp,
   rewriter.eraseOp(oldOp);
 }
 
+//===----------------------------------------------------------------------===//
+// Handshake and StdExpr Builder classes
+//===----------------------------------------------------------------------===//
+
+namespace {
+
+class HandshakeBuilder : public HandshakeVisitor<HandshakeBuilder, bool> {
+public:
+  HandshakeBuilder(ValueVectorList portList, Location insertLoc,
+                   ConversionPatternRewriter &rewriter)
+      : portList(portList), insertLoc(insertLoc), rewriter(rewriter) {}
+
+  bool visitInvalidOp(Operation *op) { return false; }
+  using HandshakeVisitor::visitHandshake;
+
+#define HANDLE(OPTYPE)                                                         \
+  bool visitHandshake(OPTYPE op) {                                             \
+    return buildOpLogic(portList, insertLoc, rewriter, &op), true;             \
+  }
```

Comment:
I think you could remove the extra indirection here and just make the buildOpLogic methods members of this new class?

Reply:
Good idea!

Reply:
Has updated.